app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,45 @@
+class MicropostsController < ApplicationController
+  before_action :logged_in_user, only: [:create, :destroy]
+  before_action :correct_user, only: :destroy
+
+  def index
+    redirect_to root_path page: params[:page]
+  end
+
+  def create
+    @micropost = current_user.microposts.build(micropost_params)
+    @micropost.image.attach(params[:micropost][:image])
+    check_for_create
+  end
+
+  def destroy
+    if @micropost.destroy
+      flash[:success] = t "micropost.delete"
+      redirect_to request.referer || root_url
+    else
+      flash[:danger] = t "micropost.delete_fail"
+    end
+  end
+
+  private
+
+  def micropost_params
+    params.require(:micropost).permit(:content, :image)
+  end
+
+  def correct_user
+    @micropost = current_user.microposts.find_by(id: params[:id])
+    redirect_to root_url if @micropost.nil?
```

Comment:
```suggestion
    @micropost = current_user.microposts.find_by(id: params[:id])
    return if @micropost
    
    flash[:error] = "KHông tìm thấy micropost" # I18n nhé
    redirect_to root_url
```
ri sẽ thân thiện người dùng